### PR TITLE
Logging: Add log levels

### PR DESCRIPTION
**Pull request by kakra**
I thought it would be nice to have some log levels in the log output because bees is very chatty. Sometimes it looks like it's writing more data by logging than it saves by deduplication. ;-)

I also thought by this little project I could learn more about the code. So here's the result, plus it makes colorful output in the systemd journal.

@Zygo You probably want to review this carefully because I wasn't able to guess all the right severity levels of each log, or their purpose. Also, I'm not quite sure what `BEESTRACE()` does, and `ChatterBox` even less. Meanwhile, at least it works for me. ;-)

Thanks.

### CODE REVIEW

**Comment by Zygo** [comment]
BEESTRACE() writes stack traces during exception handling.

ChatterBox() caches a comparison between the current file and line number and the set of enabled file and line numbers.  The idea is that you write `CRUCIBLE_CHATTER='fd.cc extentwalker.cc'` and all the CHATTER() lines in those two files will start spewing log output.

File: src/bees-context.cc [hunk 1]
```diff
@@ -825,13 +825,13 @@ BeesContext::set_root_fd(Fd fd)
...
 
-	BEESLOG("returning from set_root_fd in " << name_fd(fd));
+	BEESLOGINFO("returning from set_root_fd in " << name_fd(fd));
 }
 
 void
 BeesContext::blacklist_add(const BeesFileId &fid)
 {
-	BEESLOG("Adding " << fid << " to blacklist");
+	BEESLOGWARN("Adding " << fid << " to blacklist");
```
Comment (Zygo) [inline src/bees-context.cc]: That's INFO, maybe even DEBUG.  It's bees hiding its own files from itself.

File: src/bees-resolve.cc [hunk 2]
```diff
@@ -172,7 +172,7 @@ BeesResolver::adjust_offset(const BeesFileRange &haystack, const BeesBlockData &
 
 		// Found the hash but not the data.  Yay!
 		m_found_hash = true;
-		BEESLOG("HASH COLLISION\n"
+		BEESLOGWARN("HASH COLLISION\n"
```
Comment (Zygo) [inline src/bees-resolve.cc]: This should probably be INFO.  Some file formats use a 4K page size and CRC64 internally, and a user who has such a file will get thousands of these.

File: src/bees.h [hunk 3]
```diff
@@ -115,21 +116,25 @@ const int FLAGS_OPEN_FANOTIFY = O_RDWR | O_NOATIME | O_CLOEXEC | O_LARGEFILE;
 
 // macros ----------------------------------------
 
-#define BEESLOG(x)      do { Chatter c(BeesNote::get_name()); c << x; } while (0)
-#define BEESLOGTRACE(x) do { BEESLOG(x); BeesTracer::trace_now(); } while (0)
+#define BEESLOG(l,x)    do { Chatter c(l, BeesNote::get_name()); c << x; } while (0)
+#define BEESLOGTRACE(x) do { BEESLOG(LOG_DEBUG, x); BeesTracer::trace_now(); } while (0)
```
Comment (Zygo) [inline src/bees.h]: This gets called from `trace_now`.  The level should be ERR for consistency (it is only called during exception stack traces).

File: src/bees.h [hunk 4]
```diff
@@ -115,21 +116,25 @@ const int FLAGS_OPEN_FANOTIFY = O_RDWR | O_NOATIME | O_CLOEXEC | O_LARGEFILE;
...
 
-#define BEESTRACE(x)   BeesTracer  SRSLY_WTF_C(beesTracer_,  __LINE__) ([&]()                 { BEESLOG(x);   })
+#define BEESTRACE(x)   BeesTracer  SRSLY_WTF_C(beesTracer_,  __LINE__) ([&]()                 { BEESLOG(LOG_DEBUG, x);   })
```
Comment (Zygo) [inline src/bees.h]: This one should be LOG_ERR (see above)

**Comment by Zygo** [comment]
Comments notwithstanding, I like it.

Could you rebase this?  Sorry, I had a bunch of stuff queued up that I hadn't pushed yet.

**Comment by kakra** [comment]
Yes, I'll rebase. Hang on. :-)

File: src/bees-context.cc (hunk 1 above)
Comment (kakra) [reply src/bees-context.cc]: I chose DEBUG.

File: src/bees-resolve.cc (hunk 2 above)
Comment (kakra) [reply src/bees-resolve.cc]: Fixed as INFO.

File: src/bees.h (hunk 3 above)
Comment (kakra) [reply src/bees.h]: So, I keep this as DEBUG? At least I fixed `l` to not be confused as `1` by renaming it to `lv`.

File: src/bees.h (hunk 4 above)
Comment (kakra) [reply src/bees.h]: Done.

**Comment by kakra** [comment]
Rebased and build-tested. Since you added a few more logger calls since before the rebase, you may want to review again. I annotated the fixed ones above. Thanks.

**Pull request merged**